tests/unit/python/foglamp/services/core/scheduler/test_scheduler.py

```diff
@@ -0,0 +1,1184 @@
+        assert tasks[0].end_time is not None
+        assert tasks[0].exit_code is '0'
+
+    @pytest.mark.asyncio
+    async def test_cancel_task(self, mocker):
+        # TODO: Mandatory - Add negative tests for full code coverage
+        async def mock_process():
+            m = MagicMock()
+            m.pid = 9999
+            m.terminate = lambda: True
+            return m
+
+        async def mock_future():
+            return True
+
+        # GIVEN
+        scheduler = Scheduler()
+        scheduler._storage = Storage(core_management_host=None, core_management_port=None)
+        mocker.patch.object(scheduler, '_schedule_first_task')
+        await scheduler._get_schedules()
+
+        schedule = scheduler._ScheduleRow(
+            id=uuid.UUID("2b614d26-760f-11e7-b5a5-be2e44b06b34"),
+            process_name="North Readings to PI",
+            name="OMF to PI north",
+            type=Schedule.Type.INTERVAL,
+            repeat=datetime.timedelta(seconds=30),
+            repeat_seconds=30,
+            time=None,
+            day=None,
+            exclusive=True,
+            enabled=True)
+
+        mocker.patch.object(scheduler, '_ready', True)
+        mocker.patch.object(scheduler, '_resume_check_schedules')
+        mocker.patch.object(scheduler, '_process_scripts', return_value="North Readings to PI")
+        mocker.patch.object(scheduler, '_wait_for_task_completion', return_value=mock_future())
+        mocker.patch.object(scheduler, '_terminate_child_processes')
+        mocker.patch.object(asyncio, 'create_subprocess_exec', return_value=mock_process())
+
+        # Assert that there is no task queued for schedule
+        with pytest.raises(KeyError) as excinfo:
+            assert scheduler._schedule_executions[schedule.id] is True
+
+        # Now queue task and assert that the task has been queued
+        await scheduler.queue_task(schedule.id)
+        assert isinstance(scheduler._schedule_executions[schedule.id], scheduler._ScheduleExecution)
+
+        # Confirm that no task has started yet
+        assert 0 == len(scheduler._schedule_executions[schedule.id].task_processes)
+
+        await scheduler._start_task(schedule)
+
+        # Confirm that task has started
+        assert 1 == len(scheduler._schedule_executions[schedule.id].task_processes)
+        task_id = list(scheduler._schedule_executions[schedule.id].task_processes.keys())[0]
+
+        # Confirm that cancel request has not been made
+        assert scheduler._schedule_executions[schedule.id].task_processes[task_id].cancel_requested is None
+
+        # WHEN
+        await scheduler.cancel_task(task_id)
+
+        # THEN
+        assert scheduler._schedule_executions[schedule.id].task_processes[task_id].cancel_requested is not None
+
+    @pytest.mark.skip("_terminate_child_processes() not fit for unit test.")
+    def test__terminate_child_processes(self, mocker):
+        pass
+
+
+class Storage(StorageClient):
```

Comment:
I would suggest to 
mockStorageClient = MagicMock(spec=StorageClient) 

and 
* patch init
* patch _get_storage_service with return value of fixture dict, and similarly for query_tbl_with_payload, query_tbl
* and just make sure to assert_called_once_with for create schedule etc.